frontend/src/routes/Video.tsx

```diff
@@ -355,6 +367,26 @@ const eventFragment = graphql`
     }
 `;
 
+const playlistFragment = graphql`
+    fragment VideoPagePlaylistData on Playlist {
+        ...PlaylistBlockPlaylistData
+        __typename
+        ... on NotAllowed { dummy } # workaround
+        ... on AuthorizedPlaylist {
+            id
+            opencastId
+            title
+            description
+            entries {
+                __typename
+                ... on AuthorizedEvent { id, ...VideoListEventData }
+                ... on Missing { dummy }
+                ... on NotAllowed { dummy }
+            }
+        }
+    }
+`;
```

Comment:
I don't think all of that is necessary. Instead of using `...VideoPagePlaylistData`  in the queries above, you can just use `...PlaylistBlockPlaylistData` directly, no? I think so. The prop of `VideoPage` becomes `playlistRef: PlaylistBlockPlaylistData$key | null` then and then you can remove the `useFragment` call from `VideoPage`.